velox/functions/sparksql/window/tests/RowNumberTest.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/lib/window/tests/WindowTestBase.h"
+#include "velox/functions/sparksql/window/WindowFunctionsRegistration.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+struct RowNumberTestParam {
+  const std::string function;
+  const std::string overClause;
+};
+
+class RowNumberTestBase : public WindowTestBase {
+ protected:
+  explicit RowNumberTestBase(const RowNumberTestParam& testParam)
+      : function_(testParam.function), overClause_(testParam.overClause) {}
+
+  void testWindowFunction(const std::vector<RowVectorPtr>& vectors) {
+    WindowTestBase::testWindowFunction(vectors, function_, {overClause_});
+  }
+
+  void SetUp() override {
+    WindowTestBase::SetUp();
+    velox::functions::window::sparksql::registerWindowFunctions("");
+  }
+
+  const std::string function_;
```

Comment:
Why is this a param if its always only row_number function ?

Reply:
Also added the `nth_value`